react/features/chat/middleware.ts

```diff
@@ -573,21 +576,33 @@ function _handleReceivedMessage({ dispatch, getState }: IStore,
 
     // Provide a default for the case when a message is being
     // backfilled for a participant that has left the conference.
-    const participant = getParticipantById(state, participantId) || { local: undefined };
+    const participant = getParticipantById(state, participantId);
 
     const localParticipant = getLocalParticipant(getState);
     let _displayName, displayNameToShow;
 
     if (lobbyChat) {
         displayNameToShow = _displayName = getLobbyChatDisplayName(state, participantId);
     } else if (isFromVisitor) {
-        _displayName = getVisitorDisplayName(state, displayName);
+        _displayName = getDisplayName(state, displayName);
         displayNameToShow = `${_displayName} ${i18next.t('visitors.chatIndicator')}`;
     } else {
-        displayNameToShow = _displayName = getParticipantDisplayName(state, participantId);
+        if (!participant) {
+            displayNameToShow = _displayName = getDisplayName(state, displayName);
+
+            // source can be 'token' or 'guest'. When it is 'guest', we append a guest indicator
+            // to the display name to notify users that this message can be anything, while the token one is coming
+            // from a trusted source (the jwt token, the name can be locked from the backend if the
+            // token has the feature 'name-readonly')
+            if (source === 'guest') {
+                displayNameToShow = `${_displayName} ${i18next.t('chat.guestsChatIndicator')}`;
```

Comment:
Why do we need this here? Aren't we adding the suffix when we display the message.

Also can we use the `getDisplayNameSuffix` function here too?

Reply:
We handle it here just for the notification. I will try using the utility ... just need to split some handling ...

Reply:
Done.